src/test/java/core/Watcher.java

```diff
@@ -0,0 +1,45 @@
+package core;
+
+import io.qameta.allure.Attachment;
+import io.qameta.allure.Step;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestWatcher;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.lang.reflect.Method;
+
+public class Watcher implements TestWatcher {
+    private final WebDriver driver = WebDriverSingleton.getDriverInstance();
```

Comment:
im not sure about that line, that you will get proper dirver in multythread env, it is better to request driver on demand. im talking about lines 24, 44